traitlets/traitlets.py

```diff
@@ -489,9 +488,39 @@ def _validate(self, obj, value):
         return value
 
     def _cross_validate(self, obj, value):
-        if hasattr(obj, '_%s_validate' % self.name):
-            cross_validate = getattr(obj, '_%s_validate' % self.name)
-            value = cross_validate(value, self)
+        try:
+            cb = getattr(obj, '_%s_validate' % self.name)
+            if not callable(cb):
+                raise TraitError('A trait validator must be callable')
+        except AttributeError:
+            cb = obj._trait_validators.get(self.name,None)
+        else:
+            warn("_[traitname]_validate handlers are deprecated: use register_validator instead",
+                 DeprecationWarning, stacklevel=2)
+            if obj._trait_validators.get(self.name,None):
+                raise TraitError('Only one cross-validator is allowed: two were found')
+            return cb(value, self)
+
+        if cb:
+            if callable(cb):
+                old = getattr(obj, self.name)
+                change = {'name': self.name,
+                          'new': value,
+                          'old': old,
+                          'owner': obj}
+                nargs = len(getargspec(cb)[0])
+                if isinstance(cb, types.MethodType):
+                    nargs -= 1
+                if nargs == 0:
+                    value = cb()
```

Comment:
I don't think it makes sense to have a validator that takes no arguments.  How will it know what value to return, if it has no idea what was set?  I think validators should always take one argument, this change dictionary you created above.